go/host/rpc/clientapi/client_api_filter.go

```diff
@@ -0,0 +1,37 @@
+package clientapi
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/obscuronet/go-obscuro/go/host"
+
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/eth/filters"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/obscuronet/go-obscuro/go/host/events"
+)
+
+// FilterAPI exposes a subset of Geth's PublicFilterAPI operations.
+type FilterAPI struct {
+	host          host.Host
+	gethFilterAPI *filters.PublicFilterAPI
+}
+
+func NewFilterAPI(host host.Host, logsCh chan []*types.Log) *FilterAPI {
+	return &FilterAPI{
+		host:          host,
+		gethFilterAPI: filters.NewPublicFilterAPI(events.NewBackend(logsCh), false, 5*time.Minute),
```

Comment:
we'll have to brainstorm this a bit.

I think the "PublicFilterAPI" is something that has to be exposed only inside the enclave, and on the wallet extension (auth client).

On the host there should be something that takes in an encrypted subscription request and exposes `event.Subscription` implementations that are Obscuro specific


It's fine to use the current stuff as a stepping stone, but we have to keep this in mind.